### PR TITLE
RFC 3339 support for both Marshal and Unmarshal.

### PR DESCRIPTION
**Pull request by aren55555**
Builds upon #201.

### CODE REVIEW

**Comment by aren55555** [comment]
@quetzyg please take a quick peek and let me know what you think! This builds upon what @omarismail submitted in #201 (which I reverted yesterday in #202, due to missing the Serialization/Marshaling part, which I've now added)

File: response_test.go [hunk 1]
```diff
@@ -470,58 +471,113 @@ func TestOmitsZeroTimes(t *testing.T) {
...
-
-	data := resp.Data
-
-	if data.Attributes == nil {
-		t.Fatalf("Expected attributes")
-	}
-
-	if data.Attributes["next"] != "2016-08-17T08:27:12Z" {
-		t.Fatal("Next was not serialised into ISO8601 correctly")
+func TestMarshal_Times(t *testing.T) {
```
Comment (quetzyg) [inline response_test.go:474]: 👍🏽

**Review by quetzyg — APPROVED** [review]
Looks good, just left a few comments/suggestions.

**Comment by aren55555** [comment]
@googlebot I fixed it, @quetzyg was the author of the code and has accepted the CLA. The commits were with the email quetzy.garcia@i*******e.com which is @quetzyg.

**Comment by quetzyg** [comment]
Want me to sign CLA again @aren55555 ?

**Comment by aren55555** [comment]
@quetzyg if possible yes please, would make my life a tad bit easier! You need to sign it with quetzy.garcia@i*******e.com

**Comment by quetzyg** [comment]
@aren55555 I've added my other email, so it should be sorted now.

**Comment by aren55555** [comment]
@googlebot rescan plz

**Pull request merged**